spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -297,6 +297,9 @@ public boolean tryLock() {
 				boolean obtainedLock = this.obtainLock();
 				if (!obtainedLock) {
 					localLock.unlock();
+					if (this.reLock <= 0) {
+						removeThreadLocal();
+					}
```

Comment:
Why isn't this in the `catch` block below after the same `localLock.unlock();` ?